Sources/TuistKit/Mappers/Graph/StaticXCFrameworkModuleMapGraphMapper.swift

```diff
@@ -0,0 +1,182 @@
+import Foundation
+import Path
+import TuistCore
+import TuistSupport
+import XcodeGraph
+
+public final class StaticXCFrameworkModuleMapGraphMapper: GraphMapping {
+    private let fileHandler: FileHandling
+
+    public init(
+        fileHandler: FileHandling = FileHandler.shared
+    ) {
+        self.fileHandler = fileHandler
+    }
+
+    public func map(graph: Graph, environment: MapperEnvironment) throws -> (Graph, [SideEffectDescriptor], MapperEnvironment) {
+        let derivedDirectory = derivedDirectory(graph: graph)
+
+        var sideEffects: [SideEffectDescriptor] = []
+        var settingsToPropagate: [GraphDependency: SettingsDictionary] = [:]
```

Comment:
Why not mapping `projects` and `targets` directly instead of persisting the settings to propagate and mutating them in another traverse?

Reply:
I removed one extra pass. We still need to do two of them as we need to go through the targets topologically sorted. One extra pass shouldn't be expensive, though, so I think it's fine, although there are probably ways how to update the settings in place if the settings is computed on demand in a DFS-fashion.